### PR TITLE
Document method parameters and return types

### PR DESCRIPTION
**Pull request by Naktibalda**
### Overview
This pull request:

- [x] Adds a feature
- [x] Has tests that cover changes

### Summary
A lot of redundant `@param` and `@return` annotation had been removed after declaring correct types in Codeception.

This change documents declared types of method parameters and return types.

### Description

Effect of this change can be seen at https://github.com/Codeception/codeception.github.com/pull/646

### CODE REVIEW

**Review by DavertMik — APPROVED** [review]

**Comment by DavertMik** [comment]
This PR changes only GenMarkdownTask which we use internally in Codeception so I'm for this to be merged
@greg-1-anderson are you ok with that?

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+composer.lock
```
Comment (greg-1-anderson) [inline .gitignore]: Robo does highest / lowest testing, which requires composer.lock be committed to the repository. Committing the composer.lock also allows the phar build to be reproducible.

**Review by greg-1-anderson — CHANGES_REQUESTED** [review]
Looks good for me, except composer.lock should not be added to the .gitignore.

**Comment by DavertMik** [comment]
@greg-1-anderson can we merge it?

**Comment by DavertMik** [comment]
@greg-1-anderson I'm merging it?

**Comment by Naktibalda** [comment]
I made a small change to document parameters in correct order, when some parameters have `@param` annotations and others don't.

Effect can be seen in this commit: https://github.com/Codeception/codeception.github.com/commit/36b6d9891080afab38dcb14284b988c9614a3793

**Comment by greg-1-anderson** [comment]
LGTM. Thanks.

**Pull request merged**